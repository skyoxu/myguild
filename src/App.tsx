import { GameCanvas } from './components/GameCanvas';

function App() {
  return (
    <div className="app-container p-8 min-h-screen bg-game-ui-background">
      <header className="text-center mb-8">
        <h1 className="text-game-title font-bold text-game-primary mb-4">
          Phaser 3 + React 19 + TypeScript
        </h1>
        <p className="text-game-subtitle text-game-ui-muted">
          符合 CLAUDE.md 技术栈要求的游戏开发环境
        </p>
      </header>

      <main className="flex justify-center">
        <GameCanvas width={800} height={600} className="shadow-lg" />
      </main>

      <footer className="text-center mt-8 text-game-ui-muted">
        <p>
          🎮 技术栈：Electron + React 19 + Vite + Phaser 3 + Tailwind CSS v4 +
          TypeScript
        </p>
      </footer>
    </div>
  );
}

export default App;
