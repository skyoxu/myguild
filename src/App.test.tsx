/*
 * App 组件测试
 * 演示 React 组件的 TDD 测试方法
 */

import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import App from './app';

describe('App 组件', () => {
  describe('渲染测试', () => {
    it('应该正确渲染应用标题', () => {
      render(<App />);

      const heading = screen.getByRole('heading', { level: 1 });
      expect(heading).toBeInTheDocument();
      expect(heading).toHaveTextContent('Phaser 3 + React 19 + TypeScript');
    });

    it('应该包含正确的技术栈信息', () => {
      render(<App />);

      const title = screen.getByText(/Phaser 3.*React 19.*TypeScript/);
      expect(title).toBeInTheDocument();
    });

    it('应该包含主容器元素', () => {
      const { container } = render(<App />);

      const mainDiv = container.querySelector('div');
      expect(mainDiv).toBeInTheDocument();
    });
  });

  describe('可访问性测试', () => {
    it('应该有正确的语义结构', () => {
      render(<App />);

      // 检查标题层级
      const heading = screen.getByRole('heading', { level: 1 });
      expect(heading).toBeInTheDocument();
    });

    it('应该支持屏幕阅读器', () => {
      render(<App />);

      const heading = screen.getByRole('heading');
      expect(heading.tagName).toBe('H1');
    });
  });

  describe('结构稳定性', () => {
    it('应包含 app 根容器与关键交互入口', () => {
      render(<App />);
      const root = screen.getByTestId('app-root');
      expect(root).toBeInTheDocument();
      // 顶部标题
      expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
      // 模式切换的两个按钮（Normal / Vertical Slice），不校验具体文案，避免样式/文案抖动
      const buttons = root.querySelectorAll('button');
      expect(buttons.length).toBeGreaterThanOrEqual(2);
      // 首屏应提供开始游戏入口（未自动开始时）
      const startBtn = screen.queryByTestId('start-game');
      expect(startBtn).toBeTruthy();
    });
  });
});
