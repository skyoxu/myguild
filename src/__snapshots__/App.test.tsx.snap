// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App 组件 > 快照测试 > 应该匹配快照 1`] = `
<div>
  <h1>
    Phaser + React + TypeScript
  </h1>
</div>
`;
